src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5866,6 +5868,33 @@ public void onEviction(Sha1HashKey key, PreparedStatementHandle handle) {
             }
         }
     }
+
+    boolean isAzureDW() throws SQLServerException, SQLException {
+        if (null == isAzureDW) {
+            try (Statement stmt = this.createStatement(); ResultSet rs = stmt.executeQuery("SELECT CAST(SERVERPROPERTY('EngineEdition') as INT)");)
```

Comment:
Is there any way to avoid an extra round trip to the server to get this info?  For example, can we get any useful info from the TDS pre-login or login packets?

Reply:
I don't think we can avoid making a call to get the table metadata, but as it turns out, we were making an exact same FMTONLY call in SQLServerBulkCopy::getDestinationMetadata method (which is called as part of the bulk copy process shortly down the line). So instead of making the FMTONLY call to the database twice in the same fashion, the driver will now store the resultset from the first FMTONLY so it can be re-used in getDestinationMetadata. This means we won't be making any additional trips to the database compared to the way it was working before.